### PR TITLE
Bookmark insert/update button text

### PR DESCRIPTION
**Pull request by illia-stv**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (bookmark): Should update text of the form button whether bookmark is selected or not . Closes #000.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-bookmark/src/bookmarkui.ts [hunk 1]
```diff
@@ -339,6 +339,17 @@ export default class BookmarkUI extends Plugin {
 		} );
 	}
 
+	/**
+	 * When bookmark element is selected, form button label should be set on 'Update' otherwise it is 'Insert'.
+	 */
+	private _updateFormButtonLabel() {
+		const locale = this.editor.locale;
+		const t = locale.t;
```
Comment (pszczesniak) [inline packages/ckeditor5-bookmark/src/bookmarkui.ts:347]:
```suggestion
		const t = this.editor.locale.t;
```

File: packages/ckeditor5-bookmark/src/ui/bookmarkformview.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export default class BookmarkFormView extends View {
 	/**
 	 * The Insert button view.
```
Comment (pszczesniak) [inline packages/ckeditor5-bookmark/src/ui/bookmarkformview.ts:55]:
```suggestion
	 * The Submit button view.
```

File: packages/ckeditor5-bookmark/src/bookmarkui.ts [hunk 3]
```diff
@@ -339,6 +339,17 @@ export default class BookmarkUI extends Plugin {
 		} );
 	}
 
+	/**
+	 * When bookmark element is selected, form button label should be set on 'Update' otherwise it is 'Insert'.
+	 */
+	private _updateFormButtonLabel() {
```
Comment (pszczesniak) [inline packages/ckeditor5-bookmark/src/bookmarkui.ts:345]: In the methods where we already fire this function we already have `this._getSelectedBookmarkElement()`, maybe it would be better to pass it as a parameter (or just boolean) to not have to execute this function again?

For example:

```suggestion
	private _updateFormButtonLabel( isBookmarkSelected: boolean ) {
```

**Review by pszczesniak — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-bookmark/tests/bookmarkui.js [hunk 4]
```diff
@@ -929,6 +929,37 @@ describe( 'BookmarkUI', () => {
...
+
+		describe( 'button label', () => {
+			it( 'should have "Insert" by default', () => {
+				bookmarkUIFeature._addFormView();
+				formView = bookmarkUIFeature.formView;
+
+				expect( formView.buttonView.label ).to.equal( 'Insert' );
+			} );
+
+			it( 'should have "Insert" label when bookamrk is not selected', () => {
```
Comment (mmotyczynska) [inline packages/ckeditor5-bookmark/tests/bookmarkui.js:941]:
```suggestion
			it( 'should have "Insert" label when bookmamrk is not selected', () => {
```

File: packages/ckeditor5-bookmark/tests/bookmarkui.js [hunk 5]
```diff
@@ -929,6 +929,37 @@ describe( 'BookmarkUI', () => {
...
+
+				bookmarkUIFeature._addFormView();
+				formView = bookmarkUIFeature.formView;
+
+				bookmarkUIFeature._showUI();
+
+				expect( formView.buttonView.label ).to.equal( 'Insert' );
+			} );
+
+			it( 'should have "Update" label when bookamrk is selected', () => {
```
Comment (mmotyczynska) [inline packages/ckeditor5-bookmark/tests/bookmarkui.js:952]:
```suggestion
			it( 'should have "Update" label when bookmark is selected', () => {
```

File: packages/ckeditor5-bookmark/src/bookmarkui.ts [hunk 6]
```diff
@@ -511,6 +522,8 @@ export default class BookmarkUI extends Plugin {
 		let prevSelectedBookmark = this._getSelectedBookmarkElement();
 		let prevSelectionParent = getSelectionParent();
 
+		this._updateFormButtonLabel();
```
Comment (mmotyczynska) [inline packages/ckeditor5-bookmark/src/bookmarkui.ts:525]: Instead of calling this twice in `_startUpdatingUI()`, can we just update the button text once at the end of the `_showUI()`?

**Review by mmotyczynska — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-bookmark/src/bookmarkui.ts [hunk 7]
```diff
@@ -339,6 +339,17 @@ export default class BookmarkUI extends Plugin {
 		} );
 	}
 
+	/**
+	 * When bookmark element is selected, form button label should be set on 'Update' otherwise it is 'Insert'.
```
Comment (mmotyczynska) [inline packages/ckeditor5-bookmark/src/bookmarkui.ts:343]:
```suggestion
	 * When a bookmark element is selected, the form button label should be set to 'Update'; otherwise, it should be 'Insert'.
```

File: packages/ckeditor5-bookmark/src/bookmarkui.ts (hunk 6 above)
Comment (illia-stv) [reply packages/ckeditor5-bookmark/src/bookmarkui.ts:525]: I thought about it but found an edge case. In that case if balloon is not closed when you back to the form view the button is not updated, so we need to initialize properly the value of the button and updated properly.

[screen-capture (1).webm](https://github.com/user-attachments/assets/d53ac37b-e9c7-4d6c-9fe6-6c260731cbef)

Comment (mmotyczynska) [reply packages/ckeditor5-bookmark/src/bookmarkui.ts]: Do we have test for this edge case?

Comment (illia-stv) [reply packages/ckeditor5-bookmark/src/bookmarkui.ts]: No, but right on my way to add it

**Review by pszczesniak — APPROVED** [review]

File: packages/ckeditor5-bookmark/src/ui/bookmarkformview.ts [hunk 8]
```diff
@@ -263,7 +263,7 @@ export default class BookmarkFormView extends View {
 	 * @param eventName An event name that the `ButtonView#execute` event will be delegated to.
```
Comment (mmotyczynska) [inline packages/ckeditor5-bookmark/src/ui/bookmarkformview.ts:263]: This is no longer needed here, if you removed the function parameter.

**Review by mmotyczynska — CHANGES_REQUESTED** [review]

**Review by mmotyczynska — APPROVED** [review]
